compiler/src/Language/Bond/Codegen/CustomMapping.hs

```diff
@@ -34,46 +36,65 @@ data NamespaceMapping = NamespaceMapping
     , toNamespace :: QualifiedName      -- ^ namespace in the generated code
     }
 
-type Parser a = Parsec SourceName () a
+type Parser = Parsec Void String
 
-whitespace :: Parser String
-whitespace = many (char ' ') <?> "whitespace"
 identifier :: Parser String
-identifier = many1 (alphaNum <|> char '_') <?> "identifier"
+identifier = some (alphaNumChar <|> char '_') <?> "identifier"
+
 qualifiedName :: Parser [String]
 qualifiedName = sepBy1 identifier (char '.') <?> "qualified name"
+
+sc :: Parser ()
+sc = L.space space1 empty empty
+
 symbol :: String -> Parser String
-symbol s = whitespace *> string s <* whitespace
+symbol = L.symbol sc
+
 equal :: Parser String
 equal = symbol "="
+
+-- consume whitespace after every lexeme
+lexeme :: Parser a -> Parser a
+lexeme = L.lexeme sc
+
+decimal :: Parser Integer
+decimal = lexeme L.decimal
+
+hexadecimal :: Parser Integer
+hexadecimal = lexeme . try $ char '0' >> char' 'x' >> L.hexadecimal
+
+octal :: Parser Integer
+octal = lexeme . try $ char '0' >> char' 'o' >> L.octal
+
+natural :: Parser Integer
+natural = hexadecimal <|> octal <|> decimal
+
 integer :: Parser Integer
-integer = decimal <$> many1 digit <?> "decimal number"
-  where
-    decimal = foldl (\x d -> 10 * x + toInteger (digitToInt d)) 0
+integer = natural <|> L.signed sc decimal
```

Comment:
this should also accept signed naturals a.k.a. -0xFF and -0o17, at least that's what parsec integer did. Will fix.